Add C++ accelerometer example and use LinearFilter (#152)
wpilibsuite/frc-docs: source/docs/software/sensors/accelerometers-software.rst

            prevYAccel = yAccel;
        }

> Most accelerometers legal for FRC use are quite noisy, and it is often a good idea to combine them with the :code:`LinearFilter` class (`Java <https://first.wpi.edu/FRC/roborio/release/docs/java/edu/wpi/first/wpilibj/filters/LinearDigitalFilter.html>`__, `C++ <https://first.wpi.edu/FRC/roborio/release/docs/java/edu/wpi/first/wpilibj/filters/LinearDigitalFilter.html>`__) to reduce the noise:

Reviewer: ```suggestion
Most accelerometers legal for FRC use are quite noisy, and it is often a good idea to combine them with the :code:`LinearFilter` class (`Java <https://first.wpi.edu/FRC/roborio/release/docs/java/edu/wpi/first/wpilibj/LinearFilter.html>`__, `C++ <https://first.wpi.edu/FRC/roborio/release/docs/cpp/classfrc_1_1LinearFilter.html>`__) to reduce the noise:
```

Reviewer 2: It says those links are invalid

Reviewer: Those are the correct links on wpilib master, at least.

.. todo:: Link should be updated with correct documentation closer to kickoff when the docs website is updated

.. tabs::

    .. code-tab:: java

        Accelerometer accelerometer = new BuiltInAccelerometer();

        // Create a LinearFilter that will calculate a moving average of the measured X acceleration over the past 10 iterations of the main loop

        LinearFilter xAccelFilter = LinearFilter.movingAverage(10);

        @Override
        public void robotPeriodic() {
            // Get the filtered X acceleration
            double filteredXAccel = xAccelFilter.calculate(accelerometer.getX());
        }

    .. code-tab:: c++

        frc::BuiltInAccelerometer accelerometer;

        // Create a LinearFilter that will calculate a moving average of the measured X acceleration over the past 10 iterations of the main loop

>         auto filter = frc::LinearFilter::MovingAverage(10);

Reviewer: ```suggestion
        auto xAccelFilter = frc::LinearFilter::MovingAverage(10);
```

            prevYAccel = yAccel;
        }

> Most accelerometers legal for FRC use are quite noisy, and it is often a good idea to combine them with the :code:`LinearFilter` class (`Java <https://first.wpi.edu/FRC/roborio/release/docs/java/edu/wpi/first/wpilibj/filters/LinearDigitalFilter.html>`__, `C++ <https://first.wpi.edu/FRC/roborio/release/docs/java/edu/wpi/first/wpilibj/filters/LinearDigitalFilter.html>`__) to reduce the noise:

Reviewer 3: Note should be added to update the link to the java doc

Author: Note added in 770de7e

        // Create a LinearDigitalFilter that will calculate a moving average of the measured X acceleration over the past 10 iterations of the main loop

        LinearFilter xAccelFilter = LinearFilter.movingAverage(10);

        @Override
        public void robotPeriodic() {
            // Get the filtered X acceleration
            double filteredXAccel = xAccelFilter.calculate(accelerometer.getX());
        }

    .. code-tab:: c++

        frc::BuiltInAccelerometer accelerometer;

>         // Create a LinearDigitalFilter that will calculate a moving average of the measured X acceleration over the past 10 iterations of the main loop

Reviewer 3: ```suggestion
        // Create a LinearFilter that will calculate a moving average of the measured X acceleration over the past 10 iterations of the main loop
```

Author: Whoops, my bad for missing this. Fixed in 471f8ce

Author: My bad for missing this, fixed in 471f8ce
